File: 02-reveals_intention/1-effective_naming/1-no_ubiquitous_language/tests/users/infrastructure/in_memory_user_repository.py
import uuid
from typing import Dict, Optional

from src.users.domain.user import User
from src.users.domain.user_repository import UserRepository


class InMemoryUserRepository(UserRepository):
    def __init__(self):
        self._users: Dict[uuid.UUID, User] = {}

    def search(self, user_id: uuid.UUID) -> Optional[User]:
        return self._users.get(user_id)

    def save(self, user: User) -> None:
        self._users[user.id] = user

File: 05-tests/1-how_to_refactor_wo_breaking_tests/src/modules/users/infrastructure/persistence/InMemoryUserRepository.ts
import { User } from "../../domain/User";
import { UserRepository } from "../../domain/UserRepository";

export class InMemoryUserRepository implements UserRepository {
	private readonly users: Map<string, User> = new Map();

	async search(id: string): Promise<User | null> {
		const user = this.users.get(id);

		return user ?? null;
	}

	async save(user: User): Promise<void> {
		this.users.set(user.id, user);
	}
}
